webapp/src/sui.tsx

```diff
@@ -90,55 +91,87 @@ export class UiElement<T extends WithPopupProps> extends data.Component<T, {}> {
 }
 
 export class DropdownMenuItem extends UiElement<DropdownProps> {
+    private isOpened = false
+    private preventHide = false
+
+    private handleTab = (e: KeyboardEvent) => {
+        let charCode = (typeof e.which == "number") ? e.which : e.keyCode
+        if (charCode === 9) {
+            this.close()
+        } else if (charCode === 32 || charCode === 13) {
+            /* give the focus back to the dropdown menu, so if the menuitem opens a modal,
+               the focus will not be reset once the modal is closed. */
+            this.child("").focus()
+        }
+    }
+
+    private close() {
+        this.preventHide = false
+        this.child("").dropdown("hide")
+    }
+
     componentDidMount() {
         this.popup()
         this.child("").dropdown({
-            action: (text: string, value: any, element: any) => {
-                let htmlElement: HTMLElement
+            action: (text: string, value: any, element: JQuery) => {
+                this.close()
 
-                this.child("").dropdown("hide")
-
-                // When we use the keyboard, it is not an HTMLElement that we receive, but a JQuery. Activating click on it reproduce the same behavior than a real click.
+                // When we use the keyboard, it is not an HTMLElement that we receive, but a JQuery.
                 if (typeof element.get === "function") {
-                    let jqueryElement = element as JQuery
-                    htmlElement = jqueryElement.get(0)
-                    htmlElement.click()
-                    return
-                } else {
-                    htmlElement = element as HTMLElement
-                }
-
-                if (htmlElement.tagName.toLowerCase() === 'a') {
-                    window.open((htmlElement as HTMLLinkElement).href, '_blank')
+                    if (element.get(0).tagName.toLowerCase() === 'a') {
+                        window.open((element.get(0) as HTMLLinkElement).href, '_blank')
+                    }
                 }
             },
             fullTextSearch: true,
-            onChange: (v: string) => {
+            onChange: (v: string, text: string, item: JQuery) => {
```

Comment:
When we select an item of the menu, we give the focus to it, and prevent the dropdown to be closed.